### ISSUE TITLE
CNNs (package) loss breaks to -nan

### ISSUE THREAD

**Issue by josepllberral** [josepllberral/machine-learning-tools#4]
CNNs loss breaks sometimes when reaching convergence, returning "-nan".

**Comment by josepllberral** [comment]
Reproducible scenario:

```
library(rcnn)
data(mnist)
img_size <- c(28,28);
train <- mnist$train;
training_x <- array(train$x, c(nrow(train$x), 1, img_size)) / 255;
training_y <- binarization(train$y);
test <- mnist$test;
testing_x <- array(test$x, c(nrow(test$x), 1, img_size)) / 255;
testing_y <- binarization(test$y);
layers <- list(
    c('type' = "CONV", 'n_channels' = 1, 'n_filters' = 4, 'filter_size' = 5, 'scale' = 0.1, 'border_mode' = 'same'),
    c('type' = "POOL", 'n_channels' = 4, 'scale' = 0.1, 'win_size' = 3, 'stride' = 2),
    c('type' = "RELU", 'n_channels' = 4),
    c('type' = "FLAT", 'n_channels' = 4),
    c('type' = "LINE", 'n_visible' = 784, 'n_hidden' = 10, 'scale' = 0.1),
    c('type' = "SOFT", 'n_inputs' = 10)
);
mnist_cnn <- train.cnn(training_x, training_y, layers, batch_size = 10, training_epochs = 40, learning_rate = 1e-3, rand_seed = 1234);
```

**Comment by josepllberral** [comment]
Does NOT happens with linear networks (MLP)

**Comment by josepllberral** [comment]
After fixing "srand" bug, can't reproduce. Probably happened because bad random initialization?

**Comment by josepllberral** [comment]
Not reproducible after more than 100 iterations (error converges)
